executor/merge_join.go

```diff
@@ -165,18 +176,25 @@ func (t *mergeJoinInnerTable) reallocReaderResult() {
 
 // Close implements the Executor Close interface.
 func (e *MergeJoinExec) Close() error {
-	if err := e.baseExecutor.Close(); err != nil {
-		return errors.Trace(err)
-	}
-	return nil
+	e.memTracker.Detach()
+	e.memTracker = nil
+
+	return errors.Trace(e.baseExecutor.Close())
 }
 
 // Open implements the Executor Open interface.
 func (e *MergeJoinExec) Open(ctx context.Context) error {
 	if err := e.baseExecutor.Open(ctx); err != nil {
 		return errors.Trace(err)
 	}
+
 	e.prepared = false
+	e.memTracker = memory.NewTracker(e.id, e.ctx.GetSessionVars().MemQuotaMergeJoin)
+	e.memTracker.AttachTo(e.ctx.GetSessionVars().StmtCtx.MemTracker)
+
+	e.innerTable.memTracker = memory.NewTracker("innerTable", -1)
+	e.innerTable.memTracker.AttachTo(e.memTracker)
```

Comment:
Do we need to detach `e.innerTable.memTracker`?

Reply:
No, `innerTable.memTracker` is a child of `e.memTracker`, `e.memTracker` is detached in `Close`.